newIDE/app/src/ResourcesList/LocalPiskelBridge.js

```diff
@@ -47,16 +52,29 @@ export const openPiskel = ({
   ipcRenderer.removeAllListeners('piskel-changes-saved');
   ipcRenderer.on(
     'piskel-changes-saved',
-    (event, outputResources, newAnimationName) => {
+    (event, outputResources, newAnimationName, metadata) => {
+      const newMetadata = metadata.data
+        ? {
+            pskl: metadata,
+          }
+        : {};
+
       const resourcesManager = project.getResourcesManager();
       outputResources.forEach(resource => {
-        const imageResource = new gd.ImageResource();
         resource.name = path.relative(projectPath, resource.path); // Still needed for onChangesSaved()
-        imageResource.setFile(resource.name);
-        imageResource.setName(resource.name);
-        resourcesManager.addResource(imageResource);
-        imageResource.delete();
+        createOrUpdateResource(project, new gd.ImageResource(), resource.name);
       });
+
+      // in case this is for a tiledSprite object, save the metadata in the Image object
+      if (metadata.isTiled) {
+        resourcesManager
+          .getResource(path.relative(projectPath, outputResources[0].path))
+          .setMetadata(JSON.stringify(newMetadata));
+      } else {
+        // if not, the metadata will still be passed onto SpritesList.js to be set in the Direction object
+        outputResources[0].metadata = newMetadata;
```

Comment:
Not a big fan of attaching metadata directly to the first resource, this is a bit in contradiction with the fact that later we attach it to the Direction. Can we pass the metadata to onChangesSaved directly? (can be optional)

Reply:
It only attaches it to the first resource if isTiled is true. IsTiled is only true if a single resource is being used (called from anything else but a multiframe sprite).  I can pass and use 'singleFrame' instead if you want to?

Reply:
Ok correction here- when more than one resource is required- I attached it to the first one in order to avoid having the metadata parameter in more than one places - which might make the flow a bit bloated/confusing.
Since I am attaching it to a temporary array - it wont get saved to project.json - it will only be used for passing

I can pass metadata as another parameter and update the flow instead if you dont like it?

Reply:
ah I see what you mean. You are right it might look confusing. I will update it now